app/components/homeScreen/DailyReads.tsx

```diff
@@ -228,52 +245,88 @@ const DailyReads = ():any => {
-        games: activityDataToShow && activityDataToShow != null ? activityCategoryArrayNew[nextIndex2]?.id : 0,
-        currentadviceid: articleDataToShow && articleDataToShow != null ? articleDataToShow?.id : 0,
-        currentgamesid: activityDataToShow && activityDataToShow != null ? activityDataToShow?.id : 0,
+        advice: [articleDataToShow && articleDataToShow != null ? articleCategoryArrayNew[nextIndex] : 0,
+        articleDataToShow1 && articleDataToShow1 != null ? articleCategoryArrayNew[nextIndex + 1] : 0],
+        games: [activityDataToShow && activityDataToShow != null ? activityCategoryArrayNew[nextIndex2]?.id : 0,
+        activityDataToShow1 && activityDataToShow1 != null ? activityCategoryArrayNew[nextIndex2 + 1]?.id : 0
+        ],
+        currentadviceid: [articleDataToShow && articleDataToShow != null ? articleDataToShow?.id : 0,
+        articleDataToShow1 && articleDataToShow1 != null ? articleDataToShow1?.id : 0],
+        currentgamesid: [activityDataToShow && activityDataToShow != null ? activityDataToShow?.id : 0,
+        activityDataToShow1 && activityDataToShow1 != null ? activityDataToShow1?.id : 0
+        ],
         currentDate: DateTime.now().toISODate(),
         taxonomyid: activeChild.taxonomyData.id,
         prematureTaxonomyId: activityTaxonomyId
       };
+
       showedDailyDataCategorytoDispatch[activeChild.uuid] = { advice: advicearray, games: gamesarray }
       dispatch(setDailyArticleGamesCategory(dailyDataCategorytoDispatch));
       dispatch(setShowedDailyDataCategory(showedDailyDataCategorytoDispatch));
     } else {
-      const dailyArticles: any = [];
-      const dailyActivityList: any = [];
-      const articleDataToShow = articleData.filter((x: any) => x.id == dailyDataCategory.currentadviceid).length > 0 ?
-        articleData.filter((x: any) => x.id == dailyDataCategory.currentadviceid)[0] : null;
-      const activityDataToShow = ActivitiesData.filter((x: any) => x.id == dailyDataCategory.currentgamesid).length > 0 ?
-        ActivitiesData.filter((x: any) => x.id == dailyDataCategory.currentgamesid)[0] : null;
-      if (articleDataToShow == null && activityDataToShow == null) {
+      console.log('second time here', articleData.length);
+      const articleDataList: any = [];
+      const activityDataList: any = [];
+
+      const articleDataToShow:any = [];
+      dailyDataCategory.currentadviceid.forEach((id: any) => {
+        const filteredArticle = articleData.find((x: any) => x.id === id);
+        if (filteredArticle) {
+          articleDataToShow.push(filteredArticle);
+        }
+      });
+
+      const activityDataToShow:any = [];
+      dailyDataCategory.currentgamesid.forEach((id: any) => {
+        const filteredActivity = ActivitiesData.find((x: any) => x.id === id);
+        if (filteredActivity) {
+          activityDataToShow.push(filteredActivity);
+        }
+      });

```

Comment:
We can optimize it using filter:
`const activityDataToShow = ActivitiesData.filter(activity => dailyDataCategory.currentgamesid.includes(activity.id));`